tests/api/test_datasets.py

```diff
@@ -1,17 +1,19 @@
 import httpx
 import pytest
 
+from server.domain.common.types import id_factory
+
 
 @pytest.mark.asyncio
 async def test_dataset_crud(client: httpx.AsyncClient) -> None:
     response = await client.post("/datasets/", json={"name": "Example"})
     assert response.status_code == 201
     data = response.json()
     pk = data["id"]
-    assert isinstance(pk, int)
+    assert isinstance(pk, str)
     assert data == {"id": pk, "name": "Example"}
 
-    response = await client.get("/datasets/4242/")
+    response = await client.get(f"/datasets/{id_factory()}/")
```

Comment:
I would use a dummy id here to make it extra clear that we're expecting a 404, eg `f"/datasets/not an existing UUID/"`

Reply:
Le problème c'est qu'il y a une validation sur le param `{id}` (déclaré comme un UUID dans la route) _avant_ qu'on atteigne le code qui check la DB pour éventuellement renvoyer une 404. Autrement dit on aurait une 422 en mettant n'imp du genre `/nonexistingid`, au lieu d'une 404. Par contre je peux mettre une variable pour clarifier :

```suggestion
    non_existing_id = id_factory()
    response = await client.get(f"/datasets/{non_existing_id}/")
```

Reply:
ah oui je vois ... est-ce qu'on peut spécifiquement utiliser un uuid incorrect? C'est juste une question de principe, ne pas utiliser un "générateur correct d'uuid" pour tester une 404 (peut-être que c'est inutile comme changement, je te laisse juge)

Reply:
Tu veux dire un UUID statique valide mais qui ne correspond à rien en base ?

```suggestion
    non_existing_id = "5d80e66c-66dd-43ac-8037-af9847a64543"
    response = await client.get(f"/datasets/{non_existing_id}/")
```

Ou pouvoir passer `/whatever/` et récupérer une 404 ? Ça me semble OK, le fait qu'on renvoie actuellement une 422 est un peu une fuite d'implémentation. Faudrait que je regarde si dans la validation on peut renvoyer une 404 au lieu d'une 422 dans on "coerce" le param `{id}` en un `uuid.UUID`. L'autre option c'est d'accepter un `str` et faire la conversion nous-mêmes en renvoyant une 404 si invalide ; cette conversion doit de toute façon être faite car le domaine manipule des `UUID`.

Reply:
Par contre comme je ne suis pas là cet aprem ni demain, comme ça n'est pas bloquant on peut tout aussi bien merger en l'état puis on fera une modif / amélioration ensuite.